### ISSUE TITLE
Error translating email subject line / How to force a certain language on email templates?

### ISSUE THREAD

**Issue by ryanmwc** [craftcms/commerce#1453]
**Description**
When I trigger an order's email to be sent out from the CP, I get an error saying
```
[error][craft\commerce\services\Emails::sendEmail] Email template parse error for email “New Order (Customer)” in “Subject:”. Order: “ef4cf7f”. Template error: “An exception has been thrown during the rendering of a template ("Undefined variable: localization") in "__string_template__b7c75e377a1ca3d353056a0132784dd4d941a944e337d10943e0d186ceffc1a6" at line 1.”
```

There is no variable called `localization` used in my template. "New Order (Customer)" is the name of the email in Commerce's settings, and for the subject line I have Twig translating the text for me like:

```
{{ 'Email: New Order (Customer) Subject Line'|t({ orderNumber: order.shortNumber }) }}
```

[Here in the docs, it says](https://docs.craftcms.com/commerce/v3/order-status-emails.html#recipient):
> If “Send to custom recipient” is selected, an email address can be entered. The language of this email will be in the language of whatever user triggers the status change.

The "user" in this case is me (an admin) and I'm in the order in the CP:

<img width="759" alt="cp-send-email" src="https://user-images.githubusercontent.com/59359516/82351470-eae66f00-99fc-11ea-9b0a-999ce6c41b5c.png">

I'm guessing the error is because Commerce can't figure out which site/language to use. The docs seem to refer to when the customer on the frontend submits the order vs. when someone in the CP does it.  

1. How can I fix this error?
2. How can I set what the language should be in the email's template?

**Additional info**
- Craft version: 3.4.20
- PHP version: 7.3.9
- Commerce version: 3.1.4

**Comment by lukeholder** [comment]
We can't reproduce.

If you make the subject template a basic string like "New Order", do you still get the `localization` variable error when it renders the email?

Can't seem to find any lowercase `localization` variable in craft or commerce - are you sure it isn't being used in your template - maybe by a plugin?

**Comment by ryanmwc** [comment]
@lukeholder It was indeed my own mistake, something to do with how I set up my static translation files.  Sorry to waste your time on this! That answers question 1.

---

For anyone in the future coming across this and needing an answer to question 2 (forcing a certain language in Commerce email templates): I ended up constructing a `Site`'s handle and fetching it like this in my email Twig template:
```
{% set lang = order.orderLanguage|lower %}
{% set country = order.shippingAddress.country.iso|lower %}
{% set site = craft.app.sites.siteByHandle(lang ~ '_' ~ country) %}
```
`lang` is whatever language the customer placed the order in (which gets saved on the cart).  In my case, I'm choosing to grab the `country` off the shipping address.  All of my sites have a handle with `{language}_{country}` as their handle (like `en_de` for English in Germany), so now I can fetch and return the site by constructing the handle.

From here, I am able to force static translations in this language by passing in a 2nd `language` parameter to the `|t` filter:
```
<p>{{ 'Dear Customer'|t({ name: order.billingAddress.firstName }, language=lang) }},</p>
```

Additionally, I can query an exact entry in an "Email Templates" channel I created for my content editors to control the message. Editors can't create or delete entries in this channel, so I know the exact entry ID corresponding to the email template it'll be in, and I can get that content in the correct language by passing in the `site` I queried:
```
{% set entry = craft.entries.id(12345).site(site).one() %}
```
